rfcs/search_relevancy_sandbox/20230406-implementation_plan_update_staging_database.md

```diff
@@ -0,0 +1,166 @@
+# Implementation Plan: Update Staging Database
+
+<!-- See the implementation plan guide for more information: https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc -->
+<!-- This template is exhaustive and may include sections which aren't relevant to your project. Feel free to remove any sections which would not be useful to have. -->
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @stacimc
+- [ ] @krysal
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/392)
+- [Project Proposal]() _TBD_
+
+## Overview
+
+<!-- A brief one or two sentence overview of the implementation being described. -->
+
+This document describes how we will implement a mechanism for updating the
+staging database with the latest data from the production database.
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+The final product of this plan will be a DAG (scheduled for `@monthly`) which
+will recreate the staging database from the most recent snapshot of the
+production API database. This will be accomplished by:
+
+1. Determine the most recent automated **production** snapshot using boto3's
+   [`describe_db_snapshots`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/describe_db_snapshots.html)
+   function (an Airflow operator does not exist for this operation). This step
+   should also check the `Status` value of the response for the most recent
+   snapshot to ensure that it is `available`.
+2. In the case where the snapshot is not yet available, wait for the status of
+   the most recent snapshot using the
+   [`RdsSnapshotExistenceSensor`](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/sensors/rds/index.html#airflow.providers.amazon.aws.sensors.rds.RdsSnapshotExistenceSensor).
+3. Create a new database from this snapshot using boto3's
+   [`restore_db_instance_from_db_snapshot`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/restore_db_instance_from_db_snapshot.html)
+   function (an operator does not exist for this operation). This database will
+   be named in a way that does not conflict with the existing dev database name,
+   e.g. `dev-next-openverse-db`. Careful consideration will need to be made when
+   drafting this step to ensure that the generated database matches the settings
+   for the existing database exactly. Specifically, the following should be
+   matched:
```

Comment:
These could all be copied from a call to [`describe_db_instances`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/describe_db_instances.html) for the existing staging RDS instance, that way we don't have to maintain these settings between the infrastructure repo _and_ this new DAG.

Reply:
Excellent call!